src/audio/copier/copier.c

```diff
@@ -140,22 +159,34 @@ static int create_endpoint_buffer(struct comp_dev *parent_dev,
 	buffer->stream.valid_sample_fmt = valid_fmt;
 	buffer->buffer_fmt = copier_cfg->base.audio_fmt.interleaving_style;
 
-	if (type == ipc4_gtw_alh) {
-		struct sof_alh_configuration_blob *alh_blob;
-
+	/* For ALH multi-gateway case, configuration blob contains struct ipc4_alh_multi_gtw_cfg
+	 * with channel map and channels number for each individual gateway.
+	 */
+	if (type == ipc4_gtw_alh && is_multi_gateway(copier_cfg->gtw_cfg.node_id) &&
+	    !create_multi_endpoint_buffer) {
 		if (copier_cfg->gtw_cfg.config_length) {
-			alh_blob = (struct sof_alh_configuration_blob *)
-				copier_cfg->gtw_cfg.config_data;
-			mask = alh_blob->alh_cfg.mapping[index].channel_mask;
+			const struct sof_alh_configuration_blob *alh_blob =
+				(const struct sof_alh_configuration_blob *)
+					copier_cfg->gtw_cfg.config_data;
+			uint8_t chan_bitmask = alh_blob->alh_cfg.mapping[index].channel_mask;
+
+			chan_map = bitmask_to_nibble_channel_map(chan_bitmask);
+			buffer->stream.channels = popcount(chan_bitmask);
```

Comment:
shouldn't there be some sort of range check on the blob, specially that the map makes sense?